cpp/src/arrow/array-dict-test.cc

```diff
@@ -948,27 +948,30 @@ TEST(TestDictionary, TransposeNulls) {
   AssertArraysEqual(*expected, *out);
 }
 
-TEST(TestDictionary, DISABLED_ListOfDictionary) {
+TEST(TestDictionary, ListOfDictionary) {
   std::unique_ptr<ArrayBuilder> root_builder;
   ASSERT_OK(MakeBuilder(default_memory_pool(), list(dictionary(int8(), utf8())),
                         &root_builder));
   auto list_builder = checked_cast<ListBuilder*>(root_builder.get());
   auto dict_builder =
       checked_cast<DictionaryBuilder<StringType>*>(list_builder->value_builder());
 
+  ASSERT_EQ(dict_builder->type(), nullptr);
+  ASSERT_EQ(list_builder->type(), nullptr);
+
   ASSERT_OK(list_builder->Append());
   std::vector<std::string> expected;
-  for (char a : "abc") {
-    for (char d : "def") {
-      for (char g : "ghi") {
-        for (char j : "jkl") {
-          for (char m : "mno") {
-            for (char p : "pqr") {
+  for (char a : util::string_view("abc")) {
```

Comment:
I'm curious: does this change make a difference?

Reply:
Yes: with the raw string literal, the terminating `\0` was included in the iterated values. This way only the visible characters are iterated over